toolkit/tools/pkg/imagecustomizerlib/imagecustomizer.go

```diff
@@ -110,17 +113,47 @@ func createImageCustomizerParameters(buildDir string,
 
 	// output image
 	ic.outputImageFormat = outputImageFormat
+	ic.outputIsIso = ic.outputImageFormat == ImageFormatIso
 	ic.outputImageFile = outputImageFile
 	ic.outputImageBase = strings.TrimSuffix(filepath.Base(outputImageFile), filepath.Ext(outputImageFile))
 	ic.outputImageDir = filepath.Dir(outputImageFile)
 
-	if ic.outputImageFormat != "" && ic.outputImageFormat != ImageFormatIso {
+	if ic.outputImageFormat != "" && !ic.outputIsIso {
 		ic.qemuOutputImageFormat, err = toQemuImageFormat(ic.outputImageFormat)
 		if err != nil {
 			return nil, err
 		}
 	}
 
+	// If there are OS customizations, then we proceed as usual.
+	// If there are no OS customizations, and the input is an iso, we just
+	// return because this function is mainly about OS customizations.
+	// This function also supports shrinking/exporting partitions. While
+	// we could support those functions for input isos, we are choosing to
+	// not support them until there is an actual need/a future time.
+	// We explicitly inform the user of the lack of support here.
+	if !ic.customizeOSPartitions && ic.inputIsIso {
```

Comment:
Would it make sense to disallow `enableShrinkFilesystems` and `outputSplitPartitionsFormat` when the input is an ISO, regardless of the value of `customizeOSPartitions`?

Reply:
it does. I've updated the code.